src/Symfony/Component/Routing/Router.php

```diff
@@ -281,11 +281,10 @@ public function getMatcher()
         $class = $this->options['matcher_cache_class'];
         $baseClass = $this->options['matcher_base_class'];
         $expressionLanguageProviders = $this->expressionLanguageProviders;
-        $that = $this; // required for PHP 5.3 where "$this" cannot be use()d in anonymous functions. Change in Symfony 3.0.
 
         $cache = $this->getConfigCacheFactory()->cache($this->options['cache_dir'].'/'.$class.'.php',
-            function (ConfigCacheInterface $cache) use ($that, $class, $baseClass, $expressionLanguageProviders) {
-                $dumper = $that->getMatcherDumperInstance();
+            function (ConfigCacheInterface $cache) use ($class, $baseClass, $expressionLanguageProviders) {
```

Comment:
we no longer need to define those variables `$class, $baseClass, $expressionLanguageProviders`